code/game/objects/effects/landmarks/landmarks.dm

```diff
@@ -662,6 +683,139 @@
 	GLOB.comm_tower_landmarks_net_two -= src
 	return ..()
 
+// ammo spawn (tyrargo)
+// m41a ammo
+/obj/effect/landmark/ammo_spawn/m41a_ammo_low
+	name = "m41a ammo spawn (20)"
+	icon_state = "item_crap"
+
+/obj/effect/landmark/ammo_spawn/m41a_ammo_low/Initialize(mapload, ...)
+	. = ..()
+	if(prob(20))
+		new /obj/item/ammo_magazine/rifle(loc)
+
+/obj/effect/landmark/ammo_spawn/m41a_ammo_med
+	name = "m41a ammo spawn (30)"
+	icon_state = "ipool"
+
+/obj/effect/landmark/ammo_spawn/m41a_ammo_med/Initialize(mapload, ...)
+	. = ..()
+	if(prob(30))
+		new /obj/item/ammo_magazine/rifle(loc)
+
+/obj/effect/landmark/ammo_spawn/m41a_ammo_high
+	name = "m41a ammo spawn (60)"
+	icon_state = "item_good"
+
+/obj/effect/landmark/ammo_spawn/m41a_ammo_high/Initialize(mapload, ...)
+	. = ..()
+	if(prob(60))
+		new /obj/item/ammo_magazine/rifle(loc)
+
+// m41a ext ammo
+/obj/effect/landmark/ammo_spawn/m41a_ext_ammo_low
+	name = "m41a ext ammo spawn (20)"
+	icon_state = "item_crap"
+
+/obj/effect/landmark/ammo_spawn/m41a_ext_ammo_low/Initialize(mapload, ...)
+	. = ..()
+	if(prob(20))
+		new /obj/item/ammo_magazine/rifle/extended(loc)
+
+/obj/effect/landmark/ammo_spawn/m41a_ext_ammo_med
+	name = "m41a ext ammo spawn (30)"
+	icon_state = "ipool"
+
+/obj/effect/landmark/ammo_spawn/m41a_ext_ammo_med/Initialize(mapload, ...)
+	. = ..()
+	if(prob(30))
+		new /obj/item/ammo_magazine/rifle/extended(loc)
+
+/obj/effect/landmark/ammo_spawn/m41a_ext_ammo_high
+	name = "m41a ext ammo spawn (60)"
+	icon_state = "item_good"
+
+/obj/effect/landmark/ammo_spawn/m41a_ext_ammo_high/Initialize(mapload, ...)
+	. = ..()
+	if(prob(60))
+		new /obj/item/ammo_magazine/rifle/extended(loc)
+
+// m4ra ammo
+/obj/effect/landmark/ammo_spawn/m4ra_ammo_low
+	name = "m4ra low ammo spawn (20)"
+	icon_state = "item_crap"
+
+/obj/effect/landmark/ammo_spawn/m4ra_ammo_low/Initialize(mapload, ...)
+	. = ..()
+	if(prob(20))
+		new /obj/item/ammo_magazine/rifle/m4ra(loc)
+
+/obj/effect/landmark/ammo_spawn/m4ra_ammo_med
+	name = "m4ra med ammo spawn (40)"
+	icon_state = "ipool"
+
+/obj/effect/landmark/ammo_spawn/m4ra_ammo_med/Initialize(mapload, ...)
+	. = ..()
+	if(prob(40))
```

Comment:
All this needs refactoring to use a variable for probability rather than redefining the landmark each time

Reply:
See above.